app/routes.py

```diff
@@ -1,2 +1,231 @@
+                        completed_at=request_body["completed_at"])
+            db.session.add(new_task)
+            db.session.commit()
+            return make_response({"task": {
+                    "id": new_task.task_id,
+                    "title": new_task.title,
+                    "description": new_task.description,
+                    "is_complete": False if new_task.completed_at is None else True
+                }}, 201)
+
+    elif request.method=="GET":
+        title_query = request.args.get('sort')
+        if title_query=='desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        elif title_query=='asc':
+            tasks = Task.query.order_by(Task.title).all()
+        else:
+            tasks=Task.query.all()
+        tasks_response=[]
+        for task in tasks:
+            tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at is None else task.completed_at
+                })
+        return jsonify(tasks_response)
+
+    
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_a_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response('',404)
+
+    elif request.method == "GET":
+        if not task.goal_id:
+            return {"task": { "id": task.task_id,
+                        "title": task.title,
+                        "description": task.description,
+                        "is_complete": False if task.completed_at is None else task.completed_at  
+                        }}
+        else:
+            return {"task": { "id": task.task_id,
+                        "title": task.title,
+                        "description": task.description,
+                        "goal_id":task.goal_id,
+                        "is_complete": False if task.completed_at is None else task.completed_at  
+                        }}
+                        
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        
+        db.session.commit()
+
+        
+
+        return make_response({"task": { "id": task.task_id,
+                                        "title": task.title,
+                                        "description": task.description,
+                                        "is_complete": False if task.completed_at is None else True  
+                                        }})
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"})
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
```

Comment:
you could combine the incomplete and complete route. You could create a conditional like 
`if task.completed_at:
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked complete",
        }
    else: 
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked incomplete",
        }`